src/Background/interpolators.jl

```diff
@@ -44,3 +45,89 @@ function (z::ZoomInterpolator)(mesh::AbstractArray{T}) where T
+inverse distance weight and `leafsize` determines at what number of points to stop splitting the KDTree further.
+
+The interpolator can be called with some additional parameter being, `n_neighbors` which is the number of nearest neighbors to be considered,
+`power` is the exponent for distance in the weighing factor, `reg` is the offset for the weighing factor in denominator,
+`conf_dist` is the distance below which two points would be considered as the same point.
+
+!!! warning
+    0 < `n_neighbor` <= number of elements in tree
+
+# Examples
+```jldoctest
+julia> coordinates = [1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0 10.0; 1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0 10.0];
+
+julia> weights = [1.0 1.0 1.0 1.0 1.0 1.0 1.0 1.0 1.0 1.0];
+
+julia> value = [1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0 10.0];
+
+julia> positions = [9.9 1.0; 10.0 1.0];
+
+julia> alg = IDWInterpolator(coordinates,value, weights, 8);
+
+julia> alg(positions)
+1×2 Array{Float64,2}:
+ 9.57581  2.75894
+
+julia> IDWInterpolator(coordinates, value)(positions)
+1×2 Array{Float64,2}:
+ 9.57581  2.75894
+
+```
+"""
+struct IDWInterpolator <: BackgroundInterpolator
+    coordinates::AbstractArray
+    values::AbstractArray
+    weights::Union{AbstractArray, Nothing}
+    leafsize::Real
+
+    function IDWInterpolator(coordinates::AbstractArray, values::AbstractArray, weights::Union{AbstractArray, Nothing}, leafsize::Real)
+        leafsize <= 0 && error("Invalid leafsize=$leafsize. leafsize must be greater than 0")
+        new(coordinates, values, weights, leafsize)
+    end
+end
+
+IDWInterpolator(coordinates, value; weight = nothing, leafsize = 8) = IDWInterpolator(coordinates, value, weight, leafsize)
+
+function idw_util(alg::IDWInterpolator, idxs::AbstractArray, dist::AbstractArray, power::Real, reg::Real, conf_dist::Real)
+
+    dist[1] <= conf_dist && return alg.values[idxs[1]]
+
+    num = den = zero(eltype(dist))
+
+    for (i,j) = zip(1:length(idxs), 1:length(dist))
+        w_i = 1 / (reg + dist[i])^power
+        num += w_i * alg.values[idxs[i]] * (alg.weights === nothing ? 1 : alg.weights[idxs[i]])
+        den += w_i
+    end
+
+    den ≈ 0 && return NaN
+    return num / den
+end
+
+
+function idw_interpolator(alg::IDWInterpolator, positions::AbstractArray, n_neighbors::Real, power::Real, reg::Real, conf_dist::Real)
+    tree = KDTree(alg.coordinates, leafsize = alg.leafsize)
+
+    idxs, dist = knn(tree, positions, n_neighbors, true)
+
+    _out = zeros(eltype(dist[1]), 1, size(positions, 2))
+    for i = 1:length(_out)
+        _out[i] = idw_util(alg, idxs[i], dist[i], power, reg, conf_dist)
+    end
+
+    return _out
+end
+
+(alg::IDWInterpolator)(positions; n_neighbors = 8,
+                                  power = 1.0,
+                                  reg = 0.0,
+                                  conf_dist = 1e-12) = idw_interpolator(alg, positions, n_neighbors, power, reg, conf_dist)
```

Comment:
okay, we ~cannot~ can but shouldn't have keyword arguments in this version. The interface (set by `BackgroundInterpolator`, see the docs ;)) and `estimate_background` shows that we don't have a good method for passing keyword arguments to the estimators or background interpolators. This is why we use types for the interpolators, so that we can set all of the different settings (the `n_neighbors`, `power`, etc.) before passing it into something like `estimate_background`. 

These all have defaults, too, so just put them into the constructor for `IDWInterpolator`.